analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -473,32 +509,30 @@ def get_query(source):
 
 # %%
 if USE_SERVICEX:
-    try:
-        from func_adl_servicex import ServiceXSourceUpROOT
-    except ImportError:
-        print("cannot import func_adl_servicex, which is a required dependency when using ServiceX")
-        raise
-
+    from servicex import deliver, query, dataset
     # dummy dataset on which to generate the query
-    dummy_ds = ServiceXSourceUpROOT("cernopendata://dummy", "Events", backend_name="uproot")
-
-    # tell low-level infrastructure not to contact ServiceX yet, only to
-    # return the qastle string it would have sent
-    dummy_ds.return_qastle = True
-
-    # create the query
-    query = get_query(dummy_ds).value()
+    if USE_SERVICEX_UPROOT_RAW:
+        hl_query = get_uproot_raw_query()
+    else:
+        hl_query = get_query(query.FuncADL_Uproot())
 
     # now we query the files using a wrapper around ServiceXDataset to transform all processes at once
     t0 = time.time()
-    ds = utils.file_input.ServiceXDatasetGroup(fileset, backend_name="uproot", ignore_cache=utils.config["global"]["SERVICEX_IGNORE_CACHE"])
```

Comment:
Since I think we can fully get rid of `ServiceXDatasetGroup` now, we can also drop it from the `utils` module.